src/main/starlark/builtins_bzl/common/cc/experimental_cc_static_library.bzl

```diff
@@ -0,0 +1,301 @@
+# Copyright 2021 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This is an experimental implementation of cc_static_library.
+
+We may change the implementation at any moment or even delete this file. Do not
+rely on this.
+"""
+
+load(":common/cc/action_names.bzl", "ACTION_NAMES")
+load(":common/cc/cc_common.bzl", "cc_common")
+load(":common/cc/cc_helper.bzl", "artifact_category", "cc_helper")
+load(":common/cc/cc_info.bzl", "CcInfo")
+load(":common/cc/semantics.bzl", "semantics")
+
+cc_internal = _builtins.internal.cc_internal
+
+def _declare_static_library(*, name, actions, cc_toolchain):
+    new_name = cc_internal.get_artifact_name_for_category(
```

Comment:
Is this the only reason, cc_static_library can go directly into rules_cc?

Reply:
I can definitely send it there, but is it ready for that yet? The Unix toolchain is quite out of sync with Bazel (https://github.com/bazelbuild/rules_cc/tree/main/cc/private/toolchain) and, as far as I can tell, there is no integration test setup yet. 

I'm also not sure how to replace `cc_internal.get_artifact_name_for_category` - is `cc_internal` accessible from allowlisted repos such as `rules_cc`?

Reply:
You're right about the integration test setup, and you probably know more about how much cc toolchains are out of sync. I'd like to improve the situation.

`cc_internal.get_artifact_name_for_category` isn't accessible in any way at the moment. I'd provide something temporary for you, until we figure out the complete story of toolchain feature configuration. (Or maybe you can use some reasonable defaults?)

We had some other rules that were new and we created them directly in the depot. Examples are: py_proto_library, java_single_jar, proto_toolchain. The reason for this was, since we'd like to move the rules out of Bazel binary, anything new should be created in the repositories.